esmf_regrid/tests/__init__.py

```diff
@@ -69,3 +72,48 @@ def make_grid_args(nx, ny):
         small_grid_lon_bounds,
         small_grid_lat_bounds,
     )
+
+
+def create_temp_filename(suffix=""):
+    """
+    Return a temporary file name.
+
+    Copied from `iris.util`.
+
+    Parameters
+    ----------
+    suffix : str
+        Optional filename extension.
+
+    Returns
+    -------
+    str
+        An appropriate filename for a temporary file.
+    """
+    temp_file = tempfile.mkstemp(suffix)
+    os.close(temp_file[0])
+    return temp_file[1]
```

Comment:
@stephenworsley You're living in the wonderful world of `pytest` and there are super handy `pytest` fixtures that do this for you for free... see https://docs.pytest.org/en/6.2.x/tmpdir.html 😉 

This would remove the need for both `create_temp_filename` and `temp_filename` 👍

Reply:
I was trying to immitate iris functions here, it makes sense that pytest would be able to clear things up a bit, I'll have a look at these.